testmodule/helpers.nim

```diff
@@ -0,0 +1,53 @@
+import std/json
+import std/strutils
+import pkg/ethers
+
+proc revertReason*(e: ref EthersError): string =
+  try:
+    let json = parseJson(e.msg)
+    var msg = json{"message"}.getStr
+    const revertPrefixes = @[
+      # hardhat
+      "Error: VM Exception while processing transaction: reverted with " &
+      "reason string ",
+      # ganache
+      "VM Exception while processing transaction: revert "
+    ]
+    for prefix in revertPrefixes.items:
+      msg = msg.replace(prefix)
+    msg = msg.replace("\'")
+    return msg
+  except JsonParsingError:
+    return ""
+
+template reverts*(body: untyped): untyped =
+  let asyncproc = proc(): Future[bool] {.async.} =
+    try:
+      body
+      return false
+    except EthersError:
+      return true
+    except CatchableError:
```

Comment:
Why catch `CatchableError` here? I would expect it to be handled by the test framework like any unexpected error.

Reply:
That's fair. The purpose was to keep `reverts` and `revertsWith` as booleans as implementation becomes a bit odd. Take for example this test:

```nim
  test "only checks EthersErrors":
    let ethCall = proc() =
      raise newException(ValueError, $rpcResponse)

    check:
      doesNotRevert:
        ethCall()
```

Without catching `CatchableError`, the `ValueError` is raised and needs to be caught outside of the `check doesNotRevert`, so the usage would need to become something like:
```nim
  test "only checks EthersErrors":
    let ethCall = proc() =
      raise newException(ValueError, $rpcResponse)

    var success = false

    try:
      check:
        doesNotRevert:
          ethCall()
    except ValueError:
      success = true
    check success
```
I've updated it to remove catching `CatchableError` (and also updated so that we are checking `JsonRpcProviderErrors` instead of `EthersErrors`).